packages/mermaid/src/diagrams/xychart/xychartRenderer.ts

```diff
@@ -0,0 +1,123 @@
+import type { Diagram } from '../../Diagram.js';
+import { log } from '../../logger.js';
+import { selectSvgElement } from '../../rendering-util/selectSvgElement.js';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import type {
+  DrawableElem,
+  TextElem,
+  TextHorizontalPos,
+  TextVerticalPos,
+} from './chartBuilder/interfaces.js';
+import type XYChartDB from './xychartDb.js';
+
+export const draw = (txt: string, id: string, _version: string, diagObj: Diagram) => {
+  const db = diagObj.db as typeof XYChartDB;
+  const themeConfig = db.getChartThemeConfig();
+  const chartConfig = db.getChartConfig();
+  function getDominantBaseLine(horizontalPos: TextVerticalPos) {
+    return horizontalPos === 'top' ? 'text-before-edge' : 'middle';
+  }
+
+  function getTextAnchor(verticalPos: TextHorizontalPos) {
+    return verticalPos === 'left' ? 'start' : verticalPos === 'right' ? 'end' : 'middle';
+  }
+
+  function getTextTransformation(data: TextElem) {
+    return `translate(${data.x}, ${data.y}) rotate(${data.rotation || 0})`;
+  }
+
+  log.debug('Rendering xychart chart\n' + txt);
+
+  const svg = selectSvgElement(id);
+  const group = svg.append('g').attr('class', 'main');
+  const background = group
+    .append('rect')
+    .attr('width', chartConfig.width)
+    .attr('height', chartConfig.height)
+    .attr('class', 'background');
+
+  // @ts-ignore: TODO Fix ts errors
+  configureSvgSize(svg, chartConfig.height, chartConfig.width, true);
+
+  svg.attr('viewBox', '0 0 ' + chartConfig.width + ' ' + chartConfig.height);
```

Comment:
```suggestion
  svg.attr('viewBox', `0 0 ${chartConfig.width} ${chartConfig.height}`);
```

Reply:
Done

Reply:
I have to ask, was the suggestion about interpolation only or did it have other meaning? Can we mark it as resolved? @sidharthv96